magenta-lib/src/main/scala/magenta/tasks/ASGTasks.scala

```diff
@@ -112,8 +112,16 @@ case class CullInstancesWithTerminationTag(pkg: DeploymentPackage, stage: Stage,
     implicit val elbClient = ELB.client(keyRing, region)
     val instancesToKill = asg.getInstances.filter(instance => EC2.hasTag(instance, "Magenta", "Terminate", ec2Client))
     val orderedInstancesToKill = instancesToKill.transposeBy(_.getAvailabilityZone)
-    reporter.verbose(s"Culling instances: ${orderedInstancesToKill.map(_.getInstanceId).mkString(", ")}")
-    orderedInstancesToKill.foreach(instance => ASG.cull(asg, instance, asgClient, elbClient))
+    try {
+      reporter.verbose(s"Culling instances: ${orderedInstancesToKill.map(_.getInstanceId).mkString(", ")}")
+      orderedInstancesToKill.foreach(instance => ASG.cull(asg, instance, asgClient, elbClient))
+    } catch {
+      case e: AmazonServiceException if desiredSizeReset(e) => {
+        reporter.info(s"${e.getMessage}. Your ASG desired size may have been reset. Please check that appropriate dependencies are included in riff-raff.yaml, or ensure desiredSize isn't set in the Cloudformation.")
+      }
+    }
+
+    def desiredSizeReset(e: AmazonServiceException) = e.getStatusCode == 400 && e.getErrorCode == "ValidationError"
```

Comment:
This seems pretty vague (ASE tend to be 400 ValidationError). I tend to match on the ErrorMessage, but that's brittle. Given that we are in a given task this vagueness seems reasonable.